python/lsst/ts/criopy/gui/time_chart.py

```diff
@@ -235,7 +235,15 @@ def _topic_selected(self, obj: UnitLabel) -> None:
                 if self._signal is not None:
                     self._signal.disconnect(self._appendData)
 
-                self._createCaches({obj.unit_name: [name]})
+                try:
+                    unit_name = obj.formator.unit_name
+                except AttributeError:
+                    try:
+                        unit_name = obj.unit_name
+                    except AttributeError:
+                        unit_name = "Y"
```

Comment:
How and why? I don't see how to avoid two try-except - the only option would be to somehow arrange one try-catch into a method, but that will not increase code readability. EAFP asks even for multiple try-except blocks - see discussion at https://stackoverflow.com/questions/17015230/are-nested-try-except-blocks-in-python-a-good-programming-practice